internal/git/gitsync_processor.go

```diff
@@ -67,55 +66,79 @@ func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ s
 	}
 
 	// The revision can be a branch, a tag, or a commit hash
-	h, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
+	hash, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
 	if err != nil {
 		logger.Errorw("error resolving the revision", "revision", repo.TargetRevision, "err", err)
 		return err
 	}
 
-	// Retrieving the commit object matching the hash.
-	commit, err := r.CommitObject(*h)
-	if err != nil {
-		logger.Errorw("error checkout the commit", "hash", h.String(), "err", err)
-		return err
-	}
-
-	// Retrieve the tree from the commit.
-	tree, err := commit.Tree()
-	if err != nil {
-		logger.Errorw("error get the commit tree", "err", err)
-		return err
-	}
-
-	if !isRootDir(repo.Path) {
-		// Locate the tree with the given path.
-		tree, err = tree.Tree(repo.Path)
+	val, ok := gitSync.Status.CommitStatus[repo.Name]
+	// Only create the resources for the first time if not created yet.
+	// Otherwise, monitoring with intervals.
+	if !ok {
+		// Retrieving the commit object matching the hash.
+		commit, err := r.CommitObject(*hash)
 		if err != nil {
-			logger.Errorw("error locate the path", "err", err)
+			logger.Errorw("error checkout the commit", "hash", hash.String(), "err", err)
 			return err
 		}
-	}
 
-	// Read all the files under the path and apply each one respectively.
-	err = tree.Files().ForEach(func(f *object.File) error {
-		logger.Debugw("read file", "file_name", f.Name)
-		_, err = f.Contents()
+		// Retrieve the tree from the commit.
+		tree, err := commit.Tree()
 		if err != nil {
-			logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
+			logger.Errorw("error get the commit tree", "err", err)
 			return err
 		}
 
-		// TODO: Apply to the resources
-		return nil
-	})
-	if err != nil {
-		return err
+		if !isRootDir(repo.Path) {
+			// Locate the tree with the given path.
+			tree, err = tree.Tree(repo.Path)
+			if err != nil {
+				logger.Errorw("error locate the path", "err", err)
+				return err
+			}
+		}
+
+		// Read all the files under the path and apply each one respectively.
+		err = tree.Files().ForEach(func(f *object.File) error {
+			logger.Debugw("read file", "file_name", f.Name)
+			_, err = f.Contents()
+			if err != nil {
+				logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
+				return err
+			}
+
+			// TODO: Apply to the resources
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		commitStatus := v1.CommitStatus{
+			Hash:     hash.String(),
+			Synced:   true,
+			SyncTime: metav1.NewTime(time.Now()),
+		}
+		if gitSync.Status.CommitStatus == nil {
+			gitSync.Status.CommitStatus = make(map[string]v1.CommitStatus)
+		}
+		gitSync.Status.CommitStatus[repo.Name] = commitStatus
+		// Errors will be retried
+		if err = k8Client.Update(ctx, gitSync); err != nil {
```

Comment:
So, we need to actually be updating the Status subresource like [this](https://github.com/numaproj-labs/numaplane/blob/main/internal/controller/gitsync_controller.go#L127)

Also, when you say "Errors will be retried" in your comment, are you saying that since we have the map, we will automatically retry again the next time we come through in 3 minutes?

Reply:
Yeah, meant to use `Status()` client but cannot get the test working. Will look into that a bit more.